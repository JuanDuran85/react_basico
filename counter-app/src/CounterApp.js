import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CountApp = ({value}) => {

    const [counter, setCounter ] = useState(value);

    const incrementa = () =>{
        setCounter(counter + 1);
        // setCounter((c)=>c+1);
    };

    const decrementa = () => {
        setCounter(counter - 1);
    }

    const resetValue = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CountApp</h1>
            <p>{counter}</p>
            <button onClick={incrementa}>+1</button>
            <button onClick={resetValue}>Reset</button>
            <button onClick={decrementa}>-1</button>
        </>
    );
};

CountApp.propTypes = {
    value: PropTypes.number.isRequired,
};

export default CountApp;