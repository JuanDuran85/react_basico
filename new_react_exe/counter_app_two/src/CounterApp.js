import { PropTypes } from 'prop-types';
import { React } from 'react';

const CounterApp = ({valor}) => {

    const clickBotton = (e) => {
        console.log('click', e);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{valor}</h2>
            <button onClick={clickBotton}>+1</button>
        </>
    )
}

CounterApp.propTypes = {
    valor: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    valor: 0
}

export default CounterApp;