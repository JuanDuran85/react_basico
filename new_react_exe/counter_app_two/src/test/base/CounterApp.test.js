import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import React from "react";

import "../setupTests";
import CounterApp from "../../CounterApp";

describe("Test general texto plano", () => {
  test("debe mostrar el mensaje de Counter App", async () => {
    const wrapper = render(<CounterApp />);
    expect(wrapper.getByText("Counter App")).toBeTruthy();
  });

  test("Debe mostrar la palabra Counter App con Enzyme", () => {
    const wrapper = shallow(<CounterApp />);
    expect(wrapper.find("h1").text()).toBe("Counter App");
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el valor del contador", () => {
    const valor = 5;
    const wrapper = shallow(<CounterApp valor={valor} />);
    const textoParrafo = wrapper.find("h2").text().trim();
    expect(textoParrafo).toBe(`${valor}`);
  });
});

describe('Acciones y/o eventos', () => {
    let wrapper = shallow(<CounterApp/>);

    beforeEach(() => {
        wrapper = shallow(<CounterApp valor = { 5 }/>);
    });
    
    test('Debe incrementar el contador al hacer un click sobre el boton de +1 ', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(wrapper.find('h2').text().trim()).toBe('6');
    });

    test('Debe decrementar en uno el contador a hacer click sobre el boton de -1',()=>{
        wrapper.find('button').at(2).simulate('click');
        expect(wrapper.find('h2').text().trim()).toBe('4');
    });

    test('debe colocar el valor por defecto del contador al hacer un click sobre el boton de rest', () => {
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        expect(wrapper.find('h2').text().trim()).toBe('8');
        wrapper.find('button').at(1).simulate('click');
        expect(wrapper.find('h2').text().trim()).toBe('0');
    });
});

