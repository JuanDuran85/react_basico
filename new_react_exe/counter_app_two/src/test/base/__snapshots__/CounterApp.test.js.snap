// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test general texto plano Debe mostrar la palabra Counter App con Enzyme 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
